docs/features/role-based-access-control/high-level-design.md

```diff
@@ -0,0 +1,76 @@
+# Role base access control - High-level design
+
+This document aims at providing an overview of the architecture of Role-based Access Control (RBAC from here on) for
+the Marxan platform.
+
+## Roles
+
+**Platform admins** should be identified in a database table dedicated for the purpose. Adding users to the admins
+DB table should only be possible for other admin users.
+
+The remaining roles supported by the platform (`Owner`, `Contributor` and `Viewer`) will be stored in the `roles` 
+database in the `marxan-api` database. These roles should cover the roles defined in the project requirements:
+
+* "Platform admin" would be achieved with the implementation of a dedicated DB table for platform admins;
+* "Project owner" -> role Owner for resource Project;
+* "Project contributor" -> role Contributor for resource Project;
+* "Project reviewer (read-only)" -> role Viewer for resource Project;
+* "Project solution-viewer" -> role Viewer for resource "Solutions";
+
+There is an implicit relationship between platform actions and the supported roles. By default, it is assumed that:
+* a user with role `Viewer` can read the information for the given resource (which usually includes calls to 
+`GET resource` and `GET resource/:id` endpoints).
+* user with role `Contributor` can perform all the actions a `Viewer` can, as well as edition actions for the given
+resource (which usually includes calls to `PATCH/PUT resource/:id` endpoints).
+* user with `Owner` role can perform all the actions a `Contributor` can, as well as deleting actions for the given
+resource (which usually includes calls to `DELETE resource/:id` endpoints).
+
+**However, it is up to the functions implementing each feature (controllers, services or any other entity 
+in the context of the platform) to correctly check permissions for the action being performed.** Checking permissions
+should be done by calling the appropriate function in the Access Control service class for the resource being accessed.
+
+For example, if a user with id `1` is trying to publish a `project` with id `2`, then the `publish` function
+needs to check if that user can publish that project. This can be done by calling the `canPublish` function in the 
+access control service in the `projects-acl` module.
+
+Currently, these are the roles required to perform the following actions in the platform:
+
+| Actions / Roles | Solution Viewer | Solution Contributor | Solution Owner | Scenario Viewer | Scenario Contributor | Scenario Owner | Project Viewer | Project Contributor | Project Owner | Organization Viewer | Organization Contributor | Organization Owner | Platform admin |
```

Comment:
looking at the final shape of the permissions matrix, I think we should add a few more permissions such as _create project_ (should be available to organization contributors?), _create scenario_ (project contributors?) as well as granting of roles (it's explained in the document already but it may be useful to add it to the matrix maybe?) on projects (project owners can do so?) and organizations (organization owners?)

Reply:
Added the actions you mentioned, as well as "run Marxan" as suggested by Kamil in a comment above, thanks both :bow:
* Org owners and contributors can create projects, proj owners and contributors can create scenarios, and so on;
* Owners of the resource can manage roles of users for said resource;
* "run Marxan" can only be done by project owners and contributors.